### ISSUE TITLE
suggestion about the 'force' input argument in functions

### ISSUE THREAD

**Issue by Paul-Baudin** [stephenwhitmarsh/EpiCode#47]
I suggest this to make the use of the 'force' parameter easier : 
`output = function(cfg, force, data1, data2);`
Intstead of :  
`output = function(cfg, data1, data2, force);`

So then, when you need to load precomputed data, you can only write : 
`output = function(cfg, force);`
And you do not have to re-load input data or to search for how many empty arguments you need to put in the function before reaching 'force'.

Of course if you think it is interesting, I can help :)

**Comment by stephenwhitmarsh** [comment]
Okay, sounds good. Has to be in one big change though, so we should plan it. 

Alternatively, it could also be done by making all inputs beyond 1 (cfg) of a variable number , then figuring out at the start of the function which inputs are boolean, and go from there. E.g. when there is a boolean in one of the inputs, and it is false, and there is data saved, return that data. If it is true then force redoing it, checking if there are enough input arguments, if not exit with explanations of expected number of inputs.
In that way, thos functions will remain backwards compatible. Then the only thing that is determined is the order of data inputs, and the first being a cfg structure.

**Comment by Paul-Baudin** [comment]
Yes I agree that if we have a satisfying solution without changing the already-existing scripts, it would be great. What about adding that at the begining of functions ?

```
if nargin == 1
    fname = fullfile(cfg.datasavedir, 'data_example.mat');
    if exist(fname, 'file')
        fprintf('Reading %s\n', fname);
        load(fname);
        return;
    else
        error('No precomputed data is found, not enough input arguments to compute data');
    end
end
```
So, to load precomputed data, only write : `output = function(cfg);`

**Comment by stephenwhitmarsh** [comment]
I'm not sure that will work, because you will have to place input arguments (vargin{n}) into the expected data structures. So then it is still not clear where the 'force' variable should go, without looking for booleans, and treating the rest in series.

**Comment by Paul-Baudin** [comment]
If you 'return' before using the other input arguments (as it is the case), those arguments are not needed when calling the function. You do not have to use varargin.
As soon as you only input cfg (verified with nargin == 1), it is supposed that you want to reload data and not compute it, so no need to verify 'force' in this case ?

**Comment by stephenwhitmarsh** [comment]
are you sure? 
Do you want to give it a shot for say two functions, one without data (only cfg) and one with one or more data entries, making sure they are backwards compatible, i.e. also work with force as last argument.

**Comment by Paul-Baudin** [comment]
See #52, for an example of implementation in 1 function + with a test script.
I am pretty sure that it works, and that it does not break previous script.

**Comment by stephenwhitmarsh** [comment]
Resolved in #52
